src/store/reducers/describe.ts

```diff
@@ -106,10 +105,48 @@ const selectConsumersNames = (state: IDescribeRootStateSlice, path: string | und
 export const selectConsumers: Selector<IDescribeRootStateSlice, IConsumer[], [string | undefined]> =
     createSelector(selectConsumersNames, (names = []) => names.map((name) => ({name})));
 
-export function getDescribe({path}: {path: string}) {
-    return createApiRequest({
-        request: window.api.getDescribe({path}),
+export function getDescribe({path}: {path: string | string[]}) {
+    let request;
+    if (Array.isArray(path)) {
+        request = path.map((p) => window.api.getDescribe({path: p}));
+    } else {
+        request = window.api.getDescribe({path});
+    }
+
+    return createApiRequest<
+        typeof FETCH_DESCRIBE,
+        TEvDescribeSchemeResult,
+        IDescribeHandledlResponse
+    >({
+        request,
         actions: FETCH_DESCRIBE,
+        dataHandler: (data) => {
+            let dataPath: string | undefined;
+            let currentDescribe: IDescribeData | TEvDescribeSchemeResult = {};
+            const newData: IDescribeData = {};
+
+            if (Array.isArray(data)) {
```

Comment:
in this case `request` can be both an array and a single promise, so the typeof `data` inferred as `Response | Response[]`, and it's fine

but in case of a single request I want the type of `data` to be `Result` (and vice versa), TS should be smart enough to know. Currently this isn't the case, the type is inferred as `Result | Result[]` even for a single request

Reply:
this should be addressed in combination with this comment: https://github.com/ydb-platform/ydb-embedded-ui/pull/231#discussion_r1034701826